aptos-move/block-executor/src/view.rs

```diff
@@ -1065,14 +1217,18 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TResourceGr
         resource_tag: &Self::ResourceTag,
         _maybe_layout: Option<&Self::Layout>,
     ) -> anyhow::Result<Option<Bytes>> {
-        // TODO[agg_v2]: support layouts.
+        // TODO[agg_v2](fix): Don't see anywhere in this function where
+        // _maybe_layout input could be useful
         let mut group_read = match &self.latest_view {
             ViewState::Sync(state) => {
                 state.read_from_group(group_key, resource_tag, self.txn_idx)?
             },
-            ViewState::Unsync(state) => state
-                .unsync_map
-                .get_value_from_group(group_key, resource_tag),
+            ViewState::Unsync(state) => {
+                // TODO[agg_v2](fix): Should we push the group key into read set
+                // irresepctive of whether the read operation succeeds?
+                state.group_read_set.borrow_mut().insert(group_key.clone());
+                state.unsync_map.read_from_group(group_key, resource_tag)
+            },
         };
 
         if matches!(group_read, GroupReadResult::Uninitialized) {
```

Comment:
You need to do exchange here? in `initialize_mvhashmap_base_group_contents`?

Reply:
Basically we have a layout for one of the members, so we have to set it when we access the group

Reply:
In `initialize_mvhashmap_base_group_contents`, I set the the layout to `Unset`.  Whenever we call `read_from_group` for a resource inside a resource group, I am setting the layout of the resource based on the input. Is that okay?